cf-agent/verify_files.c

```diff
@@ -542,15 +542,9 @@ static PromiseResult VerifyFilePromise(EvalContext *ctx, char *path, const Promi
 
     if (a.haveedit)
     {
-        if (exists)
-        {
-            result = PromiseResultUpdate(result, ScheduleEditOperation(ctx, path, &a, pp));
-        }
-        else
-        {
-            RecordFailure(ctx, pp, &a, "Promised to edit '%s', but file does not exist", path);
-            result = PromiseResultUpdate(result, PROMISE_RESULT_FAIL);
-        }
```

Comment:
Maybe it would be easier to add the exception(s) here?

Reply:
Are you thinking something like this?
```C
    if (a.haveedit)
    {
        if ((StringEqual(a.template_method, "mustache") ||
             StringEqual(a.template_method, "inline_mustache")) &&
            !exists && !CreateFalseWasSpecified(pp) &&
            CfCreateFile(ctx, path, pp, &a, &result))
        {
            exists = true;
        }

        if (exists)
        {
            result = PromiseResultUpdate(result, ScheduleEditOperation(ctx, path, &a, pp));
        }
        else
        {
            RecordFailure(ctx, pp, &a, "Promised to edit '%s', but file does not exist", path);
            result = PromiseResultUpdate(result, PROMISE_RESULT_FAIL);
        }
    }
```
Not possible, because I need to know if the template rendering is successful, before creating a file when `create => "true"` is not specified.

Reply:
Something like this could work:
```C
    if (a.haveedit)
    {
        if (exists || StringEqual(a.template_method, "mustache") &&
            StringEqual(a.template_method, "inline_mustache"))
        {
            result = PromiseResultUpdate(result, ScheduleEditOperation(ctx, path, &a, pp));
        }
        else
        {
            RecordFailure(ctx, pp, &a, "Promised to edit '%s', but file does not exist", path);
            result = PromiseResultUpdate(result, PROMISE_RESULT_FAIL);
        }
    }
```
but it's very confusing

Reply:
The latter is what I had in mind. And I don't think it's confusing, esp. if you add a comment explaining what's going on. What might be confusing is that some file editing mechanisms require the file to exist and others don't. But when explained, it makes sense.

Reply:
Added comment explaining exceptions.